tools/functional-tester/etcd-tester/tester.go

```diff
@@ -35,201 +36,233 @@ func (tt *tester) runLoop() {
 				break
 			}
-			plog.Printf("[round#%d case#%d] start failure %s", i, j, f.Desc())
+			plog.Printf("%s start failure %s", prefix, f.Desc())
 
-			plog.Printf("[round#%d case#%d] start injecting failure...", i, j)
-			if err := f.Inject(tt.cluster, i); err != nil {
-				plog.Printf("[round#%d case#%d] injection error: %v", i, j, err)
-				if err := tt.cleanup(i, j); err != nil {
-					plog.Printf("[round#%d case#%d] cleanup error: %v", i, j, err)
+			plog.Printf("%s start injecting failure...", prefix)
+			if err := f.Inject(tt.cluster, round); err != nil {
+				plog.Printf("%s injection error: %v", prefix, err)
+				if err := tt.cleanup(); err != nil {
 					return
 				}
 				failed = true
 				break
 			}
-			plog.Printf("[round#%d case#%d] injected failure", i, j)
+			plog.Printf("%s injected failure", prefix)
 
-			plog.Printf("[round#%d case#%d] start recovering failure...", i, j)
-			if err := f.Recover(tt.cluster, i); err != nil {
-				plog.Printf("[round#%d case#%d] recovery error: %v", i, j, err)
-				if err := tt.cleanup(i, j); err != nil {
-					plog.Printf("[round#%d case#%d] cleanup error: %v", i, j, err)
+			plog.Printf("%s start recovering failure...", prefix)
+			if err := f.Recover(tt.cluster, round); err != nil {
+				plog.Printf("%s recovery error: %v", prefix, err)
+				if err := tt.cleanup(); err != nil {
 					return
 				}
 				failed = true
 				break
 			}
-			plog.Printf("[round#%d case#%d] recovered failure", i, j)
+			plog.Printf("%s recovered failure", prefix)
 
-			if !tt.consistencyCheck {
-				continue
-			}
 			if tt.cluster.v2Only {
-				plog.Printf("[round#%d case#%d] succeed!", i, j)
+				plog.Printf("%s succeed!", prefix)
 				continue
 			}
 
-			plog.Printf("[round#%d case#%d] canceling the stressers...", i, j)
-			for _, s := range tt.cluster.Stressers {
-				s.Cancel()
-			}
-			plog.Printf("[round#%d case#%d] canceled stressers", i, j)
-
-			plog.Printf("[round#%d case#%d] checking current revisions...", i, j)
-			var (
-				revs   map[string]int64
-				hashes map[string]int64
-				rerr   error
-				ok     bool
-			)
-			for k := 0; k < 5; k++ {
-				time.Sleep(time.Second)
-
-				revs, hashes, rerr = tt.cluster.getRevisionHash()
-				if rerr != nil {
-					plog.Printf("[round#%d case#%d.%d] failed to get current revisions (%v)", i, j, k, rerr)
-					continue
-				}
-				if currentRevision, ok = getSameValue(revs); ok {
+			if tt.consistencyCheck {
+				tt.cancelStressers()
+				failed = tt.doConsistencyCheck()
+				if failed {
 					break
 				}
-
-				plog.Printf("[round#%d case#%d.%d] inconsistent current revisions %+v", i, j, k, revs)
-			}
-			if !ok || rerr != nil {
-				plog.Printf("[round#%d case#%d] checking current revisions failed [revisions: %v]", i, j, revs)
-				if err := tt.cleanup(i, j); err != nil {
-					plog.Printf("[round#%d case#%d] cleanup error: %v", i, j, err)
-					return
-				}
-				failed = true
-				break
+				tt.startStressers()
+			} else if j == len(tt.failures)-1 {
```

Comment:
i feel we should just remove the else if and always update current revision. it is no harm anyway.